zigpy_xbee/zigbee/application.py

```diff
@@ -302,29 +302,31 @@ async def send_packet(self, packet: zigpy.types.ZigbeePacket) -> None:
                 "Cannot send a packet to a device without a known IEEE address"
             )
 
-        send_req = self._api.tx_explicit(
-            long_addr,
-            short_addr,
-            packet.src_ep or 0,
-            packet.dst_ep or 0,
-            packet.cluster_id,
-            packet.profile_id,
-            packet.radius,
-            tx_opts,
-            packet.data.serialize(),
-        )
-
-        try:
-            v = await asyncio.wait_for(send_req, timeout=TIMEOUT_TX_STATUS)
-        except asyncio.TimeoutError:
-            raise zigpy.exceptions.DeliveryError(
-                "Timeout waiting for ACK", status=TXStatus.NETWORK_ACK_FAILURE
+        async with self._limit_concurrency():
```

Comment:
Do we need both concurrency and priority limits?
The priority lock should probably do the job, and the concurrency limit does not affect all frames, only `tx_explicit`.

Reply:
We need both because they perform different roles:

 - The concurrency limit's maximum value should be set such that the XBee can continuously control devices, in a tight loop, without any requests failing due to buffer issues. Without it, sending 100 requests at once may overwhelm the XBee firmware and they will start failing. I've empirically tested the existing radio libraries (16 for zigpy-znp, 8 for bellows) and the current limit is set to the default of 8.

 - The priority lock is so that the serial port doesn't get jammed up with the 100 TX messages and unrelated commands to be put at the end of the queue instead of at the beginning. Increasing the command timeout to something extremely high like 60s also does the same job but it doesn't feel like an appropriate solution to a queuing problem.